rustls/src/error.rs

```diff
@@ -281,6 +281,7 @@ pub enum PeerMisbehaved {
     UnsolicitedServerHelloExtension,
     WrongGroupForKeyShare,
     UnsolicitedEchExtension,
+    ServerEchoedCompatibilitySessionId,
```

Comment:
I initially added this in alpha-order, but the [semver compat check job](https://github.com/cpu/rustls/actions/runs/13554255738/job/37884954276) doesn't like a new value being sliced in even though this is `non-exhaustive` because:

> This breaks downstream code that used its value via a numeric cast like `as isize`.

Reply:
Pff, not sure we should listen to it on that.

Reply:
I put it back in sorted position for now and can revisit if someone feels strongly about maintaining the numeric values.